Assets/HoloToolkit/Utilities/Scripts/ManualCameraControl.cs

```diff
@@ -0,0 +1,313 @@
+
+        rot.x += -InputCurve(mousePositionDelta.y);
+        rot.y += InputCurve(mousePositionDelta.x);
+    }
+
+    private bool ShouldMouseLook
+    {
+        get
+        {
+            // Only allow the mouse to control rotation when Unity has focus. This enables
+            // the player to temporarily alt-tab away without having the player look around randomly
+            // back in the Unity Game window.
+            if (!this.appHasFocus)
+            {
+                return false;
+            }
+            else if (this.MouseLookButton == MouseButton.None)
+            {
+                return true;
+            }
+            else if (this.MouseLookButton <= MouseButton.Middle)
+            {
+                return Input.GetMouseButton((int)this.MouseLookButton);
+            }
+            else if (this.MouseLookButton == MouseButton.Control)
+            {
+                return Input.GetKey(KeyCode.LeftControl) || Input.GetKey(KeyCode.RightControl);
+            }
+            else if (this.MouseLookButton == MouseButton.Shift)
+            {
+                return Input.GetKey(KeyCode.LeftShift) || Input.GetKey(KeyCode.RightShift);
+            }
+            else if (this.MouseLookButton == MouseButton.Focused)
+            {
+                if (!this.wasLooking)
+                {
+                    // any kind of click will capture focus
+                    return Input.GetMouseButtonDown((int)MouseButton.Left) || Input.GetMouseButtonDown((int)MouseButton.Right) || Input.GetMouseButtonDown((int)MouseButton.Middle);
+                }
+                else
+                {
+                    // pressing escape will stop capture
+                    return !Input.GetKeyDown(KeyCode.Escape);
+                }
+            }
+
+            return false;
+        }
+    }
+
+    private void OnApplicationFocus(bool focusStatus)
+    {
+        this.appHasFocus = focusStatus;
+    }
+
+    private void SetWantsMouseJumping(bool wantsJumping)
+    {
+        if (wantsJumping != this.isMouseJumping)
+        {
+            this.isMouseJumping = wantsJumping;
+
+            if (wantsJumping)
+            {
+                // unlock the cursor if it was locked
+                UnityEngine.Cursor.lockState = CursorLockMode.None;
+
+                // hide the cursor
+                UnityEngine.Cursor.visible = false;
+
+                this.lastMousePosition = Input.mousePosition;
+            }
+            else
+            {
+                // recenter the cursor (setting lockCursor has side-effects under the hood)
+                UnityEngine.Cursor.lockState = CursorLockMode.Locked;
+                UnityEngine.Cursor.lockState = CursorLockMode.None;
+
+                // show the cursor
+                UnityEngine.Cursor.visible = true;
+            }
+
+#if UNITY_EDITOR
+            UnityEditor.EditorGUIUtility.SetWantsMouseJumping(wantsJumping ? 1 : 0);
+#endif
+        }
+    }
+
+#if UNITY_EDITOR
```

Comment:
I wonder about moving this functionality out to it's own utility class. I think it'd make this class a little cleaner and more concise, and the the remote desktop check seems generally useful!